core/src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -193,13 +194,27 @@ private void createXmlTest(
       List<XmlInclude> methodNames = Lists.newArrayList(methodList.size());
       int ind = 0;
       for (ITestNGMethod m : methodList) {
-
         XmlInclude methodName =
             new XmlInclude(m.getMethodName(), m.getFailedInvocationNumbers(), ind++);
         methodName.setParameters(findMethodLocalParameters(srcXmlTest, m));
+        methodName.setConstructorOrMethod(m.getConstructorOrMethod());
         methodNames.add(methodName);
       }
-      xmlClass.setIncludedMethods(methodNames);
+
+      List<XmlInclude> groupedByMethodNames = new ArrayList<>();
+      methodNames.stream()
+              .collect(Collectors.groupingBy(XmlInclude::getConstructorOrMethod))
+              .forEach((method, methodNameList) -> {
```

Comment:
Why not using a map/collect instead of the forEach with the external collection?

Reply:
Thanks for your feedback. done